packages/export/src/watson/index.ts

```diff
@@ -23,49 +20,116 @@ export class WatsonExporter extends Exporter {
     [Botmock.SystemEntity.ORDINAL, Watson.SystemEntity.PERCENTAGE],
     [Botmock.SystemEntity.CURRENCY, Watson.SystemEntity.CURRENCY],
     [Botmock.SystemEntity.DATE, Watson.SystemEntity.DATE],
-    [Botmock.SystemEntity.DURATION, Watson.SystemEntity.TIME],
-    [Botmock.SystemEntity.COUNTRY, Watson.SystemEntity.LOCATION],
-    [Botmock.SystemEntity.NAME, Watson.SystemEntity.PERSON],
+    [Botmock.SystemEntity.DURATION, Watson.SystemEntity.TIME]
   ]);
-  /**
-   * Finds name of first intent connected to response block
-   * @param id Id of response block
-   * @param blocks {@link Botmock.Message[]}
-   */
-  #findConditionsOfBlockId = (id: string, blocks: Botmock.Message[]): string => {
-    const [intentId] = blocks
-      .flatMap(block => block.next_message_ids.filter(next => next.message_id === id))
-      .filter(next => next.intent.value)
-      .map(next => next.intent.value);
-    return this.#intents.find(intent => intent.id === intentId)?.name as string;
+
+  #getIntentName = (id: string) => {
+    const intent = this.resources.intents.filter((intent) => intent.id === id);
+    return intent[0].name;
   };
-  /**
-   * Transforms applicable block structure
-   * @param block {@link Botmock.Block}
-   */
-  #transformOutputStructure = (block: Botmock.Block): Watson.Content => {
-    const content = {} as Watson.Content;
+
+
+  //Checks for conditions attached with the Intent
+  //Input is a Path containing intent, messages array and connection_id
+  // Input Type:  intent: 'intent_id',
+  //              messages: [Array of message_ids],
+  //              connection_id: 'connection_id'
+  //Return example: `#getAge &&  @sys-number>'0' && @sys-number<'11' `
+  #getIntentConditions = (path: Watson.Path): string => {
+
+    let intent = this.resources.intents.find((intent) => intent.id === path.intent);
+       let messageBlock = this.resources.board.board.messages.flatMap(
+      (message) => message.next_message_ids
+    ).find(message => message.connection_id === path.connection_id)
+
+    let conditions;
+    if(messageBlock.conditional){
+      if(messageBlock.conditional.rules.length > 0){
```

Comment:
These two can be combined:


```suggestion
    if(messageBlock.conditional?.rules?.length > 0){
```